library-build-transformer/src/main/kotlin/io/realm/buildtransformer/RealmBuildTransformer.kt

```diff
@@ -39,123 +52,124 @@ val logger: Logger = LoggerFactory.getLogger("realm-build-logger")
  * a specific Android flavour. It is also possible to provide a list of files to delete whether or
  * not they have the annotation. These files will only be deleted from the defined flavour.
  */
-class RealmBuildTransformer(private val flavorToStrip: String,
-                            private val annotationQualifiedName: QualifiedName,
-                            private val specificFilesToStrip: Set<QualifiedName> = setOf()) : Transform() {
-
-    override fun getName(): String {
-        return "RealmBuildTransformer"
-    }
-
-    override fun isIncremental(): Boolean {
-        return true
-    }
-
-    override fun getScopes(): MutableSet<in QualifiedContent.Scope> {
-        return mutableSetOf(QualifiedContent.Scope.PROJECT)
-    }
-
-    override fun getInputTypes(): Set<QualifiedContent.ContentType> {
-        return setOf<QualifiedContent.ContentType>(QualifiedContent.DefaultContentType.CLASSES)
+class RealmBuildTransformer(
+    private val annotationQualifiedName: Property<String>,
+    private val input: ListProperty<RegularFile>,
+    private val output: RegularFileProperty,
+) {
+    init {
+        transform()
```

Comment:
Does all work really happen in the constructor? That feels a bit surprising IMO. It would maybe be nicer if there was some kind of `run` method, so we split constructing and running the class.

Reply:
I moved just exposed `transform` as `internal` and call it on the constructed object instead of implicitly from the `init`-block.